tests/pytests/unit/modules/test_aptpkg.py

```diff
@@ -1409,3 +1509,551 @@ def test_sourceslist_architectures(repo_line):
+    assert ret == {
+        "vim": {
+            "name": "vim",
+            "changes": {},
+            "result": True,
+            "comment": "Package vim is already set to be held.",
+        }
+    }
+
+
+def test_hold_pkgs():
+    """
+    test aptpkg.hold() when passing in pkgs
+    """
+    get_sel = {"hold": []}
+    mock_set_sel = MagicMock()
+    mock_set_sel.side_effect = [
+        {"vim": {"old": "install", "new": "hold"}},
+        {"vim-nox": {"old": "install", "new": "hold"}},
+    ]
+    patch_get_sel = patch("salt.modules.aptpkg.get_selections", return_value=get_sel)
+    patch_set_sel = patch("salt.modules.aptpkg.set_selections", mock_set_sel)
+    with patch_get_sel, patch_set_sel:
+        ret = aptpkg.hold(pkgs=["vim", "vim-nox"])
+        assert ret == {
+            "vim": {
+                "name": "vim",
+                "changes": {"old": "install", "new": "hold"},
+                "result": True,
+                "comment": "Package vim is now being held.",
+            },
+            "vim-nox": {
+                "name": "vim-nox",
+                "changes": {"old": "install", "new": "hold"},
+                "result": True,
+                "comment": "Package vim-nox is now being held.",
+            },
+        }
+
+
+def test_unhold():
+    """
+    test aptpkg.unhold when passing pacakge as name
+    """
+    set_sel = {"vim": {"old": "hold", "new": "install"}}
+    get_sel = {"hold": ["vim"]}
+    patch_get_sel = patch("salt.modules.aptpkg.get_selections", return_value=get_sel)
+    patch_set_sel = patch("salt.modules.aptpkg.set_selections", return_value=set_sel)
+    with patch_get_sel, patch_set_sel:
+        ret = aptpkg.unhold("vim")
+        assert ret == {
+            "vim": {
+                "name": "vim",
+                "changes": {"old": "hold", "new": "install"},
+                "result": True,
+                "comment": "Package vim is no longer being held.",
+            }
+        }
+
+
+def test_unhold_no_name_pkgs():
+    """
+    test aptpkg.unhold when we do not pass in a name or list of pkgs
+    """
+    with pytest.raises(SaltInvocationError) as err:
+        aptpkg.unhold()
+    assert err.value.message == "One of name, pkgs, or sources must be specified."
+
+
+def test_unhold_pkgs_sources():
+    """
+    test aptpkg.unhold when we we set sources and a list of pkgs.
+    """
+    with pytest.raises(SaltInvocationError) as err:
+        aptpkg.unhold(
+            pkgs=["vim", "apache2"], sources=["http://source1", "http://source2"]
+        )
+    assert err.value.message == "Only one of pkgs or sources can be specified."
+
+
+def test_unhold_sources():
+    """
+    test aptpkg.unhold when using sources
+    """
+    sources = [
+        OrderedDict(
+            [
+                (
+                    "vim",
+                    "https://mirrors.edge.kernel.org/ubuntu/pool/main/v/vim/vim_8.2.3995-1ubuntu2.12_amd64.deb",
+                )
+            ]
+        )
```

Comment:
```suggestion
        OrderedDict(
            [
                (
                    "vim",
                    "https://mirrors.edge.kernel.org/ubuntu/pool/main/v/vim/vim_8.2.3995-1ubuntu2.12_amd64.deb",
                )
            ]
        )
```
Regular dictionary? Since it's only one entry?

Reply:
I'm going to parameterize it with both, because the pkg state passes it in as an OrderedDict.